### DISCUSSION TITLE
Allow default redirect status value to be configured

### DISCUSSION THREAD

**Discussion by wout** [luckyframework/lucky#1837] [Feature requests & Ideas]
I'm using htmx in two apps now, and requests from `hx-boost` require 303 redirects after create/update/delete. Right now, I'm passing a modified status everywhere, which is tedious and error-prone. It would be great to be able to define a redirect status globally in the Lucky config.

**Comment by jwoertink** [comment]
I haven't messed with this yet, but do you use the Action's `redirect("/whatever", status: 303)` method? It looks like there's no global default right now, just default values in the methods

https://github.com/luckyframework/lucky/blob/10752adac132c79c4c0f1ed77ffd625b4e908d35/src/lucky/redirectable.cr#L121

We could make a habitat config in this module though and default it, then you could globally override it. Or did you have something in particular in mind for this?

**Reply by wout** [reply]
Yeah, I'm using `status: 303` everywhere because 302 results in an eternal redirect loop. I mentioned it on the htmx discord, and they told me 303 is the "correct" way in CREATE/PUT/DELETE situations where the result is rendered elsewhere, but that's probably also an opinion. 😏️ There's also a of discussion about the redirect status code in the Turbo/Rails community by the looks of it, so being able to configure such a thing globally wouldn't hurt.

I think a Habitat setting would be perfect, and that's exactly what I had in mind. I'm now monkey-patching the redirectable module, but that feels a bit dirty.

**Reply by jwoertink** [reply]
Cool, yeah that sounds good. We still have time to get it in the next release. We're blocked on some other stuff, so if you wanted to get that up, we can add that in.

**Reply by wout** [reply]
Great. I'll prepare a PR tomorrow!

**Comment by wout** [comment]
Implemented with https://github.com/luckyframework/lucky/pull/1838.
